### PR TITLE
fix: enoent when version set to null on tf and pulumi

### LINKED ISSUE #4832: 0.13: [Bug]: Setting the version null in pulumi provider does not work
## Garden Bonsai (0.13) Bug : Garden is not able identify the local path of pulumi when set the version to null

garden is not able to identify the pulumi when set the version to null in ```project.garden.yaml```
Below is the configuration in project.garden.yaml file,

```providers:
  - name: pulumi
    environments: ["dev"]
    # The version of pulumi to use. Set to `null` to use whichever version of `pulumi` is on your PATH.
    version: null
```

Reported Error
```An error occurred while trying to run 'pulumi/pulumi' (spawn pulumi/pulumi ENOENT). Please make sure 'pulumi/pulumi' is installed and in the $PATH.```

**With version 0.12, it detected the correct path of pulumi**
```
:information_source: x-service [debug] → Spawning '/home/workstation/.pulumi/bin/pulumi stack select telushealth/dev --create' in /home/workstation/thx/workspace/services/21062023/thxcs-shared-services/services/x/app/deployment/pulumi
```

**While with 0.13.7 version, it is not detecting the correct path of pulumi**
```
:information_source: deploy.x-service [debug] → Spawning 'pulumi/pulumi stack select telushealth/dev --create' in /home/workstation/thx/workspace/services/02062023/thxcs-shared-services/services/x/app/deployment/pulumi
```

### PR DESCRIPTION
**Pull request by Orzelius**
was caused by me in https://github.com/garden-io/garden/pull/4496

closes #4832

### CODE REVIEW

File: plugins/pulumi/cli.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export function pulumi(ctx: PluginContext, provider: PulumiProvider) {
 
 export class GlobalPulumi extends CliWrapper {
   constructor() {
-    super("pulumi", "pulumi/pulumi")
+    super({ name: "pulumi", path: "pulumi" })
```
Comment (shumailxyz) [inline plugins/pulumi/cli.ts:47]: just wondering why the change in path from `"pulumi/pulumi"` to `pulumi` unless the path was wrong earlier.

**Review by shumailxyz — APPROVED** [review]
Thanks for fixing this 🌟

File: core/src/util/ext-tools.ts [hunk 2]
```diff
@@ -54,17 +54,17 @@ export class CliWrapper {
...
   protected toolPath: string
 
-  constructor(name: string, path: string) {
+  constructor({ name, path }: { name: string; path: string }) {
     this.name = name
     this.toolPath = path
   }
 
-  async ensurePath(_: Log) {
+  async getPath(_: Log) {
```
Comment (vvagaytsev) [inline core/src/util/ext-tools.ts:62]: Looks like it doesn't have to be async.

Comment (vvagaytsev) [reply core/src/util/ext-tools.ts:62]: Ah, actually, it does. The overridden method in a sub-class is async :)

**Review by vvagaytsev — APPROVED** [review]

**Pull request merged**